### PR TITLE
feat(Clone): Add method and tests for cloning a repo resolve #7

### LINKED ISSUE #7: Clone Github repository
When a POST is received, the server should clone that branch. 

Write a function that clones a repo from URL, `clone(String url)`. It would be nice if this function returns a success code if it manages to clone the repo and a failure if it does not.

### PR DESCRIPTION
**Pull request by seccret**
- Add method for cloning repo
- Add method tearDown(...) that clears the session by closing the git session and removing the cloned repo
-  Add tests for cloning repo

### CODE REVIEW

**Comment by seccret** [comment]
Rebased to work with the spark #20 PR

File: ci/src/test/java/group10/CloneTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+public class CloneTest {
+
+  CIServer ci;
+  File cloneDirectoryPath;
+
+  @Before
+  public void setUp() {
+    ci = new CIServer();
+    cloneDirectoryPath = new File("../../gitclones");
+  }
```
Comment (jvonhacht) [inline ci/src/test/java/group10/CloneTest.java]: This cloneDirectoryPath places the test repository outside of the git repository: `DD2480_A2_CI_Group10/ci/../../gitclones`.  Tests are executed from `ci/` so perhaps change the path or use JUnit TemporaryFolder instead. JUnit can create a temp folder and clean it up after running the tests.

**Review by jvonhacht — CHANGES_REQUESTED** [review]
Good job, just one small issue with the test file path. It would be nice if we could merge this in after #18 and place the Github clone functionality inside the GithubController

File: ci/src/test/java/group10/CloneTest.java (hunk 1 above)
Comment (seccret) [reply ci/src/test/java/group10/CloneTest.java]: I changed the path

**Review by jvonhacht — APPROVED** [review]

**Pull request merged**